passes/techmap/techmap.cc

```diff
@@ -377,43 +378,43 @@ struct TechmapWorker
 			}
 		}
 
-		for (auto &it : tpl->cells_)
+		for (auto tpl_cell : tpl->cells())
 		{
-			IdString c_name = it.second->name.str();
+			IdString c_name = tpl_cell->name;
 			bool techmap_replace_cell = (!flatten_mode) && (c_name == ID::_TECHMAP_REPLACE_);
 
 			if (techmap_replace_cell)
 				c_name = orig_cell_name;
-			else if (it.second->name.begins_with("\\_TECHMAP_REPLACE_."))
+			else if (tpl_cell->name.begins_with("\\_TECHMAP_REPLACE_."))
 				c_name = stringf("%s%s", orig_cell_name.c_str(), c_name.c_str() + strlen("\\_TECHMAP_REPLACE_"));
 			else
 				apply_prefix(cell->name, c_name);
 
-			RTLIL::Cell *c = module->addCell(c_name, it.second);
+			RTLIL::Cell *c = module->addCell(c_name, tpl_cell);
 			design->select(module, c);
 
 			if (!flatten_mode && c->type.begins_with("\\$"))
 				c->type = c->type.substr(1);
 
 			vector<IdString> autopurge_ports;
 
-			for (auto &it2 : c->connections_)
+			for (auto &conn : c->connections())
 			{
 				bool autopurge = false;
 				if (!autopurge_tpl_bits.empty()) {
-					autopurge = GetSize(it2.second) != 0;
-					for (auto &bit : sigmaps.at(tpl)(it2.second))
+					autopurge = GetSize(conn.second) != 0;
+					for (auto &bit : sigmaps.at(tpl)(conn.second))
 						if (!autopurge_tpl_bits.count(bit)) {
 							autopurge = false;
 							break;
 						}
 				}
 
 				if (autopurge) {
-					autopurge_ports.push_back(it2.first);
+					autopurge_ports.push_back(conn.first);
 				} else {
-					apply_prefix(cell->name, it2.second, module);
-					port_signal_map.apply(it2.second);
+					apply_prefix(cell->name, const_cast<RTLIL::SigSpec&>(conn.second), module);
+					port_signal_map.apply(const_cast<RTLIL::SigSpec&>(conn.second));
```

Comment:
Are you cool with just `const_cast<>()`ing these?  This could also be:
```
                                       RTLIL::SigSpec new_conn = conn.second;
                                       apply_prefix(cell->name, new_conn, module);
                                       port_signal_map.apply(new_conn);
                                       c->setPort(conn.first, new_conn);
```

However, this is really an inner loop, and I'm hesitant to introduce a new copy when it's not really necessary, after all.

Reply:
No `const_cast`, please! To be honest I don't think performance is that big an issue there -- and in any case, without any hard evidence we shouldn't overthink it. Let me try.

Reply:
OK, no problem.  That's why I asked ;).